### PR TITLE
Use more idiomatic Go style

### PR DESCRIPTION
**Pull request by nmiyake**
* Use table-driven tests when possible
* Consolidate type declaration for consecutive parameters with identical types
* Use embedded struct with inline YML marshalling for LauncherConfig to share
  common data and validation logic
* Fix minor typos/touch up comments
* Use "want/got" nomenclature in tests
* Use testify for test assertionsi and requirements

---
This change is [Reviewable](https://reviewable.io/reviews/palantir/go-java-launcher/16)

### CODE REVIEW

**Comment by nmiyake** [comment]
None of this is strictly necessary, but general suggestions. If you only want to take a subset of these feel free to comment.

File: launchlib/launcher_test.go [hunk 1]
```diff
@@ -121,27 +104,16 @@ func TestKeysAreNotExpanded(t *testing.T) {
...
-		}
+	if got, ok := env["{{CWD}}"]; ok {
+		assert.Equal(t, "Value", got, "%%CWD%% environment variable incorrect")
 	} else {
 		t.Errorf("Expected %%CWD%% to exist in map and not be expanded, but it didn't")
 	}
 }
 
-func (m *mockProcessExecutor) Exec(command string, args []string, env []string) error {
+func (m *mockProcessExecutor) Exec(command string, args, env []string) error {
```
Comment (uschi2000) [inline launchlib/launcher_test.go:114]: wow, this is idiomatic? looks really strange :(

Comment (uschi2000) [reply launchlib/launcher_test.go:114]: (to an outsider)

File: launchlib/config.go [hunk 2]
```diff
@@ -20,51 +20,56 @@ import (
...
-	Classpath     []string
-	JvmOpts       []string `yaml:"jvmOpts"`
-	Args          []string
+	LauncherConfig `yaml:",inline"`
+	ServiceName    string            `yaml:"serviceName"`
+	MainClass      string            `yaml:"mainClass"`
+	JavaHome       string            `yaml:"javaHome"`
+	Env            map[string]string `yaml:"env"`
+	Classpath      []string
+	JvmOpts        []string `yaml:"jvmOpts"`
```
Comment (uschi2000) [inline launchlib/config.go]: indentation/alignment is off

File: launchlib/config_test.go [hunk 3]
```diff
@@ -67,67 +75,57 @@ env:
...
+			want: CustomLauncherConfig{
+				LauncherConfig: LauncherConfig{
+					ConfigType:    "java",
+					ConfigVersion: 1,
+				},
+				Env: map[string]string{
+					"SOME_ENV_VAR": "{{CWD}}/etc/profile",
+				},
+				JvmOpts: []string{"jvmOpt1", "jvmOpt2"}},
 		},
```
Comment (uschi2000) [inline launchlib/config_test.go:130]: alignment is off

**Review by uschi2000 — COMMENTED** [review]
thanks, Nick. some weirdness around indentation, otherwise I'm fine

File: launchlib/launcher_test.go (hunk 1 above)
Comment (nmiyake) [reply launchlib/launcher_test.go:114]: I guess I can't definitively say whether it's idiomatic, but in my experience consecutive parameters with the same type in a function definition are almost always condensed such that the declaration only occurs after the last one:

https://tour.golang.org/basics/5
https://github.com/a8m/go-lang-cheat-sheet#functions

File: launchlib/config.go (hunk 2 above)
Comment (nmiyake) [reply launchlib/config.go]: This is the alignment written/enforced by `gofmt` (Go's standard formatting tool).

Alignment can be made to look more natural by either moving declaration of `JvmOpts` before `ClassPath` or by adding `yaml` definitions for `ClassPath` and `Args` for consistency (would recommend the latter)

File: launchlib/config_test.go (hunk 3 above)
Comment (nmiyake) [reply launchlib/config_test.go:130]: Don't think alignment is actually off here, but will switch to expanded style to make this more clear.

**Comment by uschi2000** [comment]
Ok, if that's the expected style, then feel free to keep it.

**Comment by uschi2000** [comment]
+1

**Comment by nmiyake** [comment]
Great, thanks! Feel free to merge whenever you feel comfortable with it.

**Pull request merged**